napari/layers/utils/stack_utils.py

```diff
@@ -286,22 +286,68 @@ def images_to_stack(images: list[Image], axis: int = 0, **kwargs) -> Image:
     if not images:
         raise IndexError(trans._('images list is empty', deferred=True))
 
+    if not all(isinstance(layer, Image) for layer in images):
+        raise ValueError(
+            trans._(
+                'All selected layers to be merged must be Image layers.',
+                deferred=True,
+            )
+        )
+
     data, meta, _ = images[0].as_layer_data_tuple()
 
-    kwargs.setdefault('scale', np.insert(meta['scale'], axis, 1))
-    kwargs.setdefault('translate', np.insert(meta['translate'], axis, 0))
+    # RGB images do not need extra dimensions inserted into metadata
+    if 'rgb' not in kwargs:
+        kwargs.setdefault('scale', np.insert(meta['scale'], axis, 1))
+        kwargs.setdefault('translate', np.insert(meta['translate'], axis, 0))
 
     meta.update(kwargs)
-    meta['units'] = (pint.get_application_registry().pixel,) + meta['units']
-    meta['axis_labels'] = (f'axis -{data.ndim + 1}',) + meta['axis_labels']
     new_data = np.stack([image.data for image in images], axis=axis)
+
+    # RGB images do not need extra dimensions inserted into metadata
+    # They can use the meta dict from one of the source image layers
+    if not meta['rgb']:
+        meta['units'] = (pint.get_application_registry().pixel,) + meta[
+            'units'
+        ]
+        meta['axis_labels'] = (f'axis -{data.ndim + 1}',) + meta['axis_labels']
+
     return Image(new_data, **meta)
 
 
 def merge_rgb(images: list[Image]) -> Image:
     """Variant of images_to_stack that makes an RGB from 3 images."""
     if not (len(images) == 3 and all(isinstance(x, Image) for x in images)):
         raise ValueError(
-            trans._('merge_rgb requires 3 images layers', deferred=True)
+            trans._(
+                'Merging to RGB requires exactly 3 Image layers', deferred=True
+            )
         )
-    return images_to_stack(images, axis=-1, rgb=True)
+    if not all(image.data.shape == images[0].data.shape for image in images):
+        raise ValueError(
+            trans._(
+                'Shape mismatch! To merge to RGB, all selected Image layers (with R, G, and B colormaps) must have the same shape.'
```

Comment:
Also think that it will the valuable to add information about shapes that are passed. 

Otherwise, people may have problems understanding if they have images with dummy (`(1, x, y)` and `(x, y)` images, for example).